omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/resource/IntegrityRuleCategoryResource.java

```diff
@@ -0,0 +1,104 @@
+        throw new ResourceDoesNotSupportOperationException();
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+        if (representation instanceof FullRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("ruleName");
+            description.addProperty("ruleCategory");
+            description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+            description.addSelfLink();
+            return description;
+        } else if (representation instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("ruleName");
+            description.addProperty("ruleCategory");
+            description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+            description.addSelfLink();
+            return description;
+        } else if (representation instanceof RefRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("ruleName");
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+
+    private DataIntegrityService getDataIntegrityService() {
+        return Context.getService(DataIntegrityService.class);
+    }
+
+    @Override
+    public DataIntegrityRule getParent(DataIntegrityRule dataIntegrityRule) {
+        return dataIntegrityRule;
+    }
+
+    @Override
+    public void setParent(DataIntegrityRule dataIntegrityRule, DataIntegrityRule dataIntegrityRule2) {
+        dataIntegrityRule.setHandlerClassname(dataIntegrityRule2.getHandlerClassname());
+        dataIntegrityRule.setHandlerConfig(dataIntegrityRule.getHandlerConfig());
+        dataIntegrityRule.setRuleId(dataIntegrityRule.getRuleId());
+        dataIntegrityRule.setResults(dataIntegrityRule.getResults());
+        dataIntegrityRule.setRuleCategory(dataIntegrityRule.getRuleCategory());
+        dataIntegrityRule.setRuleName(dataIntegrityRule.getRuleName());
+        dataIntegrityRule.setId(dataIntegrityRule.getId());
+        dataIntegrityRule.setUuid(dataIntegrityRule.getUuid());
+    }
+
+    @Override
+    public PageableResult doGetAll(DataIntegrityRule dataIntegrityRule, RequestContext requestContext) throws ResponseException {
+        String ruleCategory = requestContext.getParameter("ruleCategory");
+        return new NeedsPaging<>(getDataIntegrityService().getRuleByCategory(ruleCategory), requestContext);
```

Comment:
The Problem is, Could not list the output from the `getRuleByCategory()`. So you should create the list of DataIntegrityRule before paging.
Can you try this method to implement the listable,
```
List<DataIntegrityRule> dataIntegrityRule = (ArrayList<DataIntegrityRule>)getDataIntegrityService().getRuleByCategory(ruleCategory);
return new NeedsPaging<>( dataIntegrityRule , requestContext);
```

Reply:
@suthagar23 your suggestion is awesome! 👍